plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/GlobalURIEditorOpener.java

```diff
@@ -73,7 +76,10 @@ public IEditorPart open(URI uri, boolean select) {
 	public IEditorPart open(URI referenceOwnerURI, EReference reference, int indexInList, boolean select) {
 		IResourceServiceProvider resourceServiceProvider = resourceServiceProviderRegistry.getResourceServiceProvider(referenceOwnerURI.trimFragment());
 		if(resourceServiceProvider instanceof IResourceUIServiceProvider) {
-			return ((IResourceUIServiceProvider) resourceServiceProvider).getURIEditorOpener().open(referenceOwnerURI, reference, indexInList, select);
+			IURIEditorOpener editorOpener = ((IResourceUIServiceProvider) resourceServiceProvider).getURIEditorOpener();
+			if (editorOpener != null) {
+				return editorOpener.open(referenceOwnerURI, reference, indexInList, select);
+			}
 		}
 		return openDefaultEditor(referenceOwnerURI, reference, indexInList, select);
 	}
```

Comment:
The two modifications highlight that the lines are basically the same; would it be worth taking the opportunity to extract the logic in a method?